### ISSUE TITLE
Does not work with Rec Room Studio

### ISSUE THREAD

**Issue by PrimeOnVR** [convalise/unity-material-symbols#1]
You see, i would love to use this plugin in my projects, but i use a modified version of unity built for the platform known as Rec Room. All i ask is for you to add a "convert to image" or "convert to font" button on the material symbol script, as Rec Room Studio does not allow you to upload external scripts/components to their platform. Thanks, and hopefully you can get this sorted so i can use the plugin, no matter what project i may be in!

_TLDR: for me to be able to use this package it needs to be convertible to an image/text mesh pro component!_

![image](https://github.com/user-attachments/assets/b5a7c30b-77bd-472a-913f-f55eeb3b597c)

**Comment by PrimeOnVR** [comment]
note: i am not aware that this does not use TMP, but it doesnt really matter as all i am asking is a way to convert it into a default format.

Rec Room Studio does have basic unity text support.

**Comment by convalise** [comment]
Hello!
Forgive me, I didn't quite understand your issue. I'll try addressing your main points separately, and hopefully, I'll be able to shed some light on the matter:

> convert to image

This package is based on the vectorized font file and not on the image files. If you need actual images, your best bet is to download the icons as PNGs directly from Google's official website: https://fonts.google.com/icons.

> convert to font

As I mentioned above, this package is already built on top of Google's font file, so I'm not sure what you mean, since the font file is already there.

> it needs to be convertible to text mesh pro component
> note: i am now aware that this does not use TMP, but it doesn't really matter as all i am asking is a way to convert it into a default format.

This package already uses a default format, as the main class inherits from `UnityEngine.UI.Text`. Regarding TMP, you're right; I made the decision not to inherit from it, as I explained on the front page. However, you can use TMP indirectly. I have provided a simple method in the [FAQ](https://github.com/convalise/unity-material-symbols?tab=readme-ov-file#6-can-i-use-the-package-with-textmeshpro) that you can try. Let me know if it helps!

**Comment by PrimeOnVR** [comment]
Let me rephrase this.

Due to how the modified version of unity I use works, when I upload my work, it cannot include any custom scripts. However, I can work with them on my client. So what I am asking is that, since you use a custom script that is placed on the object to control properties, for me to be able to use your plugin, there must be a way to convert the object back into being a text object with the Unicode character and settings already set from whatever is entered into the inspector script pre conversion.

TLDR: I need to be able to configure an icon, and then have a button to turn it into a Text component (without any custom scripts) with any settings I have applied.

**Comment by convalise** [comment]
Create a new script and do something like this:

```cs
public MaterialSymbolData icon;
public Text text;

public void OnValidate()
{
    text.text = icon.code.ToString();
}
```

It will grant you the icon picker and the unicode will be set to the linked Text component. Then you'll be able to configure the Text component through its inspector to fulfil your needs (setting the font to one of the TTFs provided in the package, size, color, etc).

**Comment by PrimeOnVR** [comment]
Thanks! Is there any place that I need to put this script specifically? I will be trying it as soon as I get unity open.

**Comment by convalise** [comment]
I see. No worries, here's the full script you can copy and paste to a new file on your project. 

```cs
using UnityEngine;
using UnityEngine.UI;
using com.convalise.UnityMaterialSymbols;

public class MaterialSymbolSetterUtil : MonoBehaviour
{
    public MaterialSymbolData icon;
    public Text text;

    public void OnValidate()
    {
        text.text = icon.code.ToString();
    }
}
```

Name the file "MaterialSymbolSetterUtil.cs" and place it anywhere in your project. I recommend you create a "Scripts" folder for better organization (it could be "Assets/Scripts", you name it).

So step 1: create a default UI Text on your scene and configure it on the inspector like this:
- set the font to the "MaterialSymbols-Standard" or the "MaterialSymbols-Filled" provided in the package;
- set the rect transform width and height to 100 x 100 (or any size you want);
- set the font size to 100 (or any size you want);
- set the alignment to be center-center
- set the horizontal and vertical overflow to "overflow";
- set the color to white

Step 2: attach the script to any game object on the scene and link the Text you have just configured.

Step 3: use the icon inspector as usual, any icon you select will be set to the Text component you've linked.

And that's it. It's OK if this script gets ignored in the build because all it's doing is setting the unicode to the Text object during editor time. You can even delete it after setting the icon if you want.

**Comment by PrimeOnVR** [comment]
Perfect! it works great! only two things i need to ask however:

1: how do i increase resolution beyond 300?
(if it is not possible, how would i convert this all to textmeshpro?)

2: how could i make an object in the Create menu in the editor to spawn this with everything preconfigured?
(for consistency with icons, of course)

i wont close the issue just yet so i can get these important things figured out. After that, i will probably be using this quite commonly (as it simplifies my workflow heavily)

**Comment by PrimeOnVR** [comment]
yo, i made the github!

might not be as up to date as yours but its made for Rec Room Studio / or places you cannot upload scripts

check it out!

https://github.com/PrimeOnVR/Material-Symbols-RRS

**Comment by convalise** [comment]
I'm glad you got it sorted out.
